src/app/utils/subnetIpRange.ts

```diff
@@ -0,0 +1,88 @@
+import type { Subnet } from "../store/subnet/types";
+
+/**
+ * Takes a subnet CIDR notation (IPv4) and returns the first and last IP of the subnet.
+ * The network and host addresses are excluded.
+ *
+ * @param cidr The CIDR notation of the subnet
+ * @returns The first and last valid IP addresses as two strings in a list.
+ */
+export const getIpRangeFromCidr = (cidr: Subnet["cidr"]) => {
+  // https://gist.github.com/binarymax/6114792
+
+  // Get start IP and number of valid addresses
+  const startIp = cidr.split("/")[0];
+  const numberOfAddresses = (1 << (32 - parseInt(cidr.split("/")[1]))) - 1;
```

Comment:
```suggestion
  const [startIp, identifier] = cidr.split("/");
  const numberOfAddresses = (1 << (32 - parseInt(identifier))) - 1;
```

Reply:
Yeah, that's definitely better tbh. I'll refactor

Reply:
I've refactored it to follow this style, but I've used "mask" instead of "identifier" since it's more accurate to the vocabulary of subnets